src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/facebook/FacebookSettingsPageObject.java

```diff
@@ -64,14 +66,19 @@ public void removeAppIfPresent() {
           }
         }
       }
+      PageObjectLogging.log("removeApp", "Wikia App removed", true);
     }
-    PageObjectLogging.log("removeApp", "Wikia App removed", true);
   }
 
   /**
    * This method verifies if App is present on facebook apps list. It searches for Wikia string.
    */
   private boolean isAppPresent() {
+    try {
+      Thread.sleep(3000);
```

Comment:
Ok, if there really is no such determining element (that will be there for certain), then it is ok. Just make sure the 3000 ms time is the best choice. As you can see we wan't to minimize false negatives as much as possible, so that the tests are stable and reliable. And the set sleep periods often create a risk for the stability of a test.